conformance/base/manifests.yaml

```diff
@@ -19,7 +19,7 @@ metadata:
   name: same-namespace
   namespace: gateway-conformance-infra
 spec:
-  gatewayClassName: {GATEWAY_CLASS_NAME}
+  gatewayClassName: "{GATEWAY_CLASS_NAME}"
```

Comment:
We're still in a position where we need to patch an annotation onto the `Deployments` using `kubectl --kustomize`; however, this was previously always getting mangled as an object instead of a string. It's just a placeholder that gets replaced by the test runner [here](https://github.com/kubernetes-sigs/gateway-api/blob/5a69157e5ca3e7831834af4ca27e3f6ac26db612/conformance/utils/kubernetes/apply.go#L57).